src/main/java/org/myrobotlab/sensor/TimeEncoder.java

```diff
@@ -79,8 +85,93 @@
 
   boolean enabled = true;
 
+  static class Positions implements Runnable {
+
+    static Positions instance = null;
+    Map<String, Double> positions = new HashMap<>();
+    transient private Thread worker;
+    transient boolean running = false;
+    String filename = null;
+    // reference counter - when 0 shuts thread down
+    int refCount;
+
+    public Positions() {
+      // load previous positions
+      String positionsDir = Service.getDataDir(Servo.class.getSimpleName());
+      filename = positionsDir + File.separator + "positions.json";
+      String json = null;
+      try {
+        json = FileIO.toString(filename);
+      } catch (Exception e) {
+      }
+      if (json != null) {
+        positions = CodecUtils.fromJson(json, HashMap.class);
+      }
+    }
+
+    @Override
+    public void run() {
+
+      running = true;
+
+      while (running) {
+        try {
+          Thread.sleep(2000);
+          log.debug("saving {} positions of {} servos", filename, positions.size());
+          FileIO.toFile(filename, CodecUtils.toJson(positions).getBytes());
```

Comment:
This seems very heavy handed to me.  Why would we bother with this?  On startup , servos go to their rest position at half speed.  The only time we should save the positions/state of any service should be when we explicitly tell the service to save its state.

Reply:
a future enhancement should be something that skips writing the update if the servos haven't moved since the last time...